addons/udes_stock/tests/test_scanned_by.py

```diff
@@ -0,0 +1,66 @@
+from .common import BaseUDES
+from odoo.api import Environment
+
+
+class TestScannedBy(BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        """Setup a picking to test user scanned by is correct"""
+        super().setUpClass()
+        # Setup user
+        cls.test_company = cls.create_company("test_company")
+        cls.test_user = cls.create_user(
+            "test_user",
+            "test_user_login",
+            company_id=cls.test_company.id,
+            company_ids=[(6, 0, cls.test_company.ids)],
+        )
+
+        cls.other_user = cls.create_user(
+            "other_user",
+            "other_user_login",
+            company_id=cls.test_company.id,
+            company_ids=[(6, 0, cls.test_company.ids)],
+        )
+
+        test_user_env = Environment(cls.env.cr, uid=cls.test_user.id, context=cls.env.context)
+
+        # Setup picking
+        cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+        cls.create_quant(cls.banana.id, cls.test_stock_location_02.id, 20)
+        picking = cls.create_picking(cls.picking_type_internal)
+        move = cls.create_single_move(cls.apple, 10, picking)
+        move = cls.create_single_move(cls.banana, 20, picking)
+        picking.action_confirm()
+        picking.action_assign()
+
+        cls.picking = picking.with_env(test_user_env)
+
+    # TODO: Does this need enabling even with real_uid removed now?
```

Comment:
Yes this is still needed, we still want to be sure that `sudo` calls do not change who completed the operation.